manifests/agent.pp

```diff
@@ -37,13 +40,15 @@
   Boolean $service_enable = true,
   Hash $config_hash = {},
   Array[Sensu::Backend_URL] $backends = ['localhost:8081'],
+  Optional[Boolean] $show_diff = undef,
```

Comment:
The `Optional` makes this more complicated than it should be. Let's just use `Boolean` with the default set to `true`, which is the default anyhow for the file resource.

https://puppet.com/docs/puppet/6.1/types/file.html#file-attribute-show_diff

Reply:
Updated to default to `true` and documented that changing agent password should set `show_diff` to `false`.